assets/js/modules/tagmanager/datastore/versions.js

```diff
@@ -398,6 +399,38 @@ const baseSelectors = {
 			);
 		}
 	),
+
+	/**
+	 * Determines whether the live container version has finished loading.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} TRUE if the GTM module is not available or the live container version has finished loading, otherwise FALSE.
+	 */
+	hasFinishedLoadingContainers: createRegistrySelector( ( select ) => () => {
```

Comment:
This selector should be moved to the analytics datastore as IB suggests, because if the Tag Manager module is not activated, this selector won't exist and `SettingsForm`/`SetupForm` will throw an exception.

Reply:
@eugene-manuilov, initially, I added this selector in the Analytics datastore, as you can see in this [commit](https://github.com/google/site-kit-wp/pull/6444/commits/4a83ca0f65e9406ce696d699f67ea543627dd509). However, later I decided to move it to Tagmanager as it makes more sense to have it there.

> this selector won't exist and SettingsForm/SetupForm will throw an exception.

It won't throw any exception since I'm returning `true` early from the selector if GTM is not connected.
https://github.com/google/site-kit-wp/blob/8cee30c0cb6eff967ead18dc22154940e14fc2ae/assets/js/modules/tagmanager/datastore/versions.js#L411-L415

Reply:
@hussain-t, the problem is that the tagmanager bundle is loaded only if the Tag Manager module is activated, it is disabled, then there will be no `hasFinishedLoadingContainers` available at all and your calls to it will throw an exception. Please, move this selector to the Analytics module.

https://github.com/google/site-kit-wp/blob/f1439fb4281d1d4be098136283f7e4f322cf1b2b/includes/Modules/Tag_Manager.php#L532-L540

Reply:
@eugene-manuilov, I have moved it to the Analytics store and renamed it to `hasFinishedLoadingGTMContainers` since it makes more sense as it is in the Analytics store now.

By the way, I did a little experimentation to verify if it throws an exception if the GTM isn't connected. However, it didn't throw any exceptions.

**When GTM is not connected but Analytics is connected**

![Screenshot 2023-01-26 at 11 05 28 AM](https://user-images.githubusercontent.com/24408261/214766456-a1a922a7-08e8-43b3-b070-96c5ad5d6b9c.png)

**When both the modules aren't connected**

![Screenshot 2023-01-26 at 11 06 21 AM](https://user-images.githubusercontent.com/24408261/214766540-c188b03a-88a8-4091-97de-cf64f89603eb.png)

Reply:
Thanks, @hussain-t. That's weird because it wasn't supposed to work. Anyway, let's better keep it in the Analytics module anyway.